### PR TITLE
Feature: Compact tab list (from sba but better)

### PR DESCRIPTION
**Pull request by CalMWolfs**
- Added compact tab list like in SBA
- Fixes a few minor issues with the SBA tab list
    - God pot timer not working
    - Not filtering long string when having no dungeon blessings
- Added the ability to remove any strings that contain HYPIXEL.NET
- Renamed both GUI render events to make more sense

### CODE REVIEW

**Comment by CalMWolfs** [comment]
Making a draft until 0.20 releases then I'll fix up the merge conflicts

File: src/main/java/at/hannibal2/skyhanni/features/misc/compacttablist/RenderColumn.kt [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        lines.add(line)
+    }
+
+    fun getMaxWidth(): Int {
+        var maxWidth = 0
+        for (tabLine in lines) {
+            maxWidth = maxWidth.coerceAtLeast(tabLine.getWidth())
+        }
+        return maxWidth
+    }
```
Comment (lineargraph) [inline src/main/java/at/hannibal2/skyhanni/features/misc/compacttablist/RenderColumn.kt]:
```suggestion
    fun getMaxWidth(): Int {
	return lines.maxOfOrNull { tabLine.getWidth() } ?: 0
    }
```

File: src/main/java/at/hannibal2/skyhanni/features/misc/compacttablist/TabColumn.kt [hunk 2]
```diff
@@ -0,0 +1,23 @@
+package at.hannibal2.skyhanni.features.misc.compacttablist
+
+
+import java.util.*
+
+class TabColumn(title: String) {
+
+    var columnTitle = title
+    val lines: MutableList<String> = LinkedList()
+    val sections: MutableList<TabSection> = LinkedList<TabSection>()
```
Comment (lineargraph) [inline src/main/java/at/hannibal2/skyhanni/features/misc/compacttablist/TabColumn.kt]: is there a reason specifically a linkedlist is chosen (over `mutableListOf()`)

**Review by lineargraph — APPROVED** [review]

**Review by lineargraph — APPROVED** [review]

**Review by hannibal002 — APPROVED** [review]

File: src/main/java/at/hannibal2/skyhanni/config/features/MiscConfig.java [hunk 3]
```diff
@@ -535,6 +535,25 @@ public static class GlowingDroppedItems {
...
+    public static class CompactTabList {
+        @Expose
+        @ConfigOption(name = "Enabled", desc = "Compacts the tablist to make it look much nicer like SBA did. Also " +
+                "doesn't break god-pot detection and shortens some other lines.") //made tablist one word here so both searches will pick it up
+        @ConfigEditorBoolean
+        @FeatureToggle
+        public boolean enabled = false;
+
+        @Expose
+        @ConfigOption(name = "Hide Hypixel Adverts", desc = "Hides text from advertising the Hypixel server or store in the tablist")
```
Comment (Absterge) [inline src/main/java/at/hannibal2/skyhanni/config/features/MiscConfig.java:552]: Add a period after tablist

**Pull request merged**